lesson02-variables-datatype/src/main/java/Exercise/Bai04.java

```diff
@@ -0,0 +1,53 @@
+package Exercise;
+
+import java.util.Random;
+
+public class Bai04 {
+
+	public static void main(String[] args) {
+		int[] listNumber = new int[9];
+		Random rd = new Random();
+		for (int i = 0; i < 9; i++) {
+			listNumber[i] = rd.nextInt(50);
+		}
+		for (int i = 0; i < listNumber.length; i++) {
+			System.out.print(listNumber[i] + " ");
+		}
+		System.out.println("-->" + oddNumberSecond(listNumber));
+		
+
+	}
+
+	public static int oddNumberSecond(int[] a) {
+		int lenghtArrayOther = 0;
+		for (int i = 0; i < a.length; i++) {
+			if (a[i] % 2 != 0) {
+				lenghtArrayOther++;
+			}
+		}
+
+		int[] arrayOther = new int[lenghtArrayOther];
+		int i = 0;
+		while (i < arrayOther.length) {
+			for (int j = 0; j < a.length; j++) {
+				if (a[j] % 2 != 0) {
+					arrayOther[i] = a[j];
+					i++;
+				}
+			}
+		}
+
+		int tamp = 0;
+		for (int j = 0; j < arrayOther.length - 1; j++) {
+			for (int j2 = 0; j2 < arrayOther.length - 1 - j; j2++) {
+				if (arrayOther[j2] < arrayOther[j2 + 1]) {
+					tamp = arrayOther[j2];
+					arrayOther[j2] = arrayOther[j2 + 1];
+					arrayOther[j2 + 1] = tamp;
+				}
+			}
+		}
+		return arrayOther[1];
```

Comment:
please validate random number array in 3 cases:
1. Array do not have any odd number (e.g [2, 4, 6, 8])
2. Array exists only ONE odd number (e.g [2, 4, 7, 10])
3. Array contains multiple duplicate max odd elements (e.g [2, 6, 9 , 9, 7, 0 , 0])

Reply:
Using sort descending and then get the second odd number is very good way
But, you have to create 2 array and few loop structure. Sort algorithm permutates elements so many times
You can think about the idea of the way to find max, min number in array and adapt in in this case
Snippet code
`int firstMax, secondMax;

if (numbers[0] > numbers[1]) {
	firstMax = numbers[0];
	secondMax = numbers[1];
} else {
	firstMax = numbers[1];
	secondMax = numbers[0];
}

for (int i = 0; i < numbers.length; i++) {
	int number = numbers[i];
	if (number % 2 != 0 && number > secondMax) {
		if (number > firstMax) {
			secondMax = firstMax;
			firstMax = number;
		} else {
			secondMax = number;
		}
	}
}
`